src/thinblock.cpp

```diff
@@ -164,78 +177,22 @@ bool CThinBlock::process(CNode *pfrom, int nSizeThinBlock)
-                bool inMissingTx = mapMissingTx.count(hash) > 0;
-                bool inOrphanCache = mapOrphanTransactions.count(hash) > 0;
-
-                if ((inMemPool && inMissingTx) || (inOrphanCache && inMissingTx))
-                    unnecessaryCount++;
-
-                if (inOrphanCache)
-                {
-                    tx = mapOrphanTransactions[hash].tx;
-                    setUnVerifiedOrphanTxHash.insert(hash);
-                }
-                else if (inMemPool && fXVal)
-                    setPreVerifiedTxHash.insert(hash);
-                else if (inMissingTx)
-                    tx = mapMissingTx[hash];
-            }
-            if (tx.IsNull())
-                missingCount++;
-
-            // In order to prevent a memory exhaustion attack we track transaction bytes used to create Block
-            // to see if we've exceeded any limits and if so clear out data and return.
-            uint64_t nTxSize = RecursiveDynamicUsage(tx);
-            uint64_t nCurrentMax = 0;
-            if (maxAllowedSize >= nTxSize)
-                nCurrentMax = maxAllowedSize - nTxSize;
-            if (thindata.AddThinBlockBytes(nTxSize, pfrom) > nCurrentMax)
-            {
-                LogPrint("thin", "thin block too large %lu %llu %llu\n", vTxHashes.size(), nTxSize,
-                    pfrom->nLocalThinBlockBytes);
-                LEAVE_CRITICAL_SECTION(cs_xval); // maintain locking order with vNodes
-                if (ClearLargestThinBlockAndDisconnect(pfrom))
-                {
-                    ENTER_CRITICAL_SECTION(cs_xval);
-                    return error("Thinblock has exceeded memory limits of %ld bytes", maxAllowedSize);
-                }
-                ENTER_CRITICAL_SECTION(cs_xval);
-            }
-            if (pfrom->nLocalThinBlockBytes > nCurrentMax)
-            {
-                LogPrint("thin", "node %s xthin block is too large %lu %llu %llu\n", pfrom->GetLogName(),
-                    vTxHashes.size(), nTxSize, pfrom->nLocalThinBlockBytes);
-                thindata.ClearThinBlockData(pfrom);
-                pfrom->fDisconnect = true;
-                return error("This thinblock has exceeded memory limits of %ld bytes", maxAllowedSize);
-            }
+        if (!ReconstructBlock(pfrom, fXVal, missingCount, unnecessaryCount))
+            return false;
 
-            // This will push an empty/invalid transaction if we don't have it yet
-            pfrom->thinBlock.vtx.push_back(tx);
-        }
         pfrom->thinBlockWaitingForTxns = missingCount;
         LogPrint("thin", "Thinblock %s waiting for: %d, unnecessary: %d, txs: %d full: %d\n",
             pfrom->thinBlock.GetHash().ToString(), pfrom->thinBlockWaitingForTxns, unnecessaryCount,
-            pfrom->thinBlock.vtx.size(), mapMissingTx.size());
+            pfrom->thinBlock.vtx.size(), pfrom->mapMissingTx.size());
     } // end lock cs_orphancache, mempool.cs, cs_xval
```

Comment:
Can you please reword this whole log line.  It is not clear to a casual reader what they mean.  Can I suggest the words "unknown", "known" and "total" (or "block").  Not clear what full means.